sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -255,16 +263,15 @@ public ChatThreadAsyncClient getChatThreadClient(String chatThreadId) {
 
     /**
      * Receive real-time notifications.
-     * @param skypeUserToken the skype user token
      * @param context the Android app context
      * @throws RuntimeException if real-time notifications failed to start.
      */
-    public void startRealtimeNotifications(String skypeUserToken, Context context) {
+    public void startRealtimeNotifications(Context context) {
         if (this.signalingClient.hasStarted()) {
             return;
         }
 
-        this.signalingClient.start(skypeUserToken, context);
+        this.signalingClient.start(context);
     }
```

Comment:
The idea here is to store the credential instead of asking for skypeUserToken every time.

For Push Notification, for both start/stop, a valid skypeUserToken is required. So I updated the ChatAsyncClient to store the credential and request for token when it's required. With this update, ChatAsyncClient would expose following functions for realtime/push notification:
public void startRealtimeNotifications(Context context)
public void stopRealtimeNotifications()
public void startPushNotifications(String deviceRegistrationToken)
public void stopPushNotifications()

But since this is a breaking change and we can't do it for stable release. I think I would revert the interface change for realtime notification and expose the following:
public void startRealtimeNotifications(String skypeUserToken, Context context)
public void stopRealtimeNotifications()
public void startPushNotifications(String skypeUserToken, String deviceRegistrationToken)
public void stopPushNotifications(String skypeUserToken)

What do you think?

Reply:
If you think users won't be passing a `skypeUserToken` in the future, you could add overloads that don't include said parameter instead. //cc @JonathanGiles in case he thinks we should do something else here.

Reply:
Do you mean leave 
public void startRealtimeNotifications(String skypeUserToken, Context context)
there and add an extra
public void startRealtimeNotifications(Context context)
?

Reply:
I forget, has this library already GA'd? If so, we can't remove existing API, and the code in your previous comment is accurate.. If it has not yet GA'd, then it is fine to remove it.

Reply:
Got it. Yes, this API is GA'd. I'll add an overload for it instead of update it. Thanks! @JonathanGiles